pkg/provider/openshiftversionprovider.go

```diff
@@ -0,0 +1,85 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/blang/semver/v4"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kong/kubernetes-telemetry/pkg/types"
+)
+
+const (
+	// OpenShiftVersionKey is a report key used to report the OpenShift version, if any.
+	OpenShiftVersionKey = types.ProviderReportKey("openshift_version")
+	// OpenShiftVersionKind is the OpenShift cluster version kind.
+	OpenShiftVersionKind = Kind(OpenShiftVersionKey)
+
+	// OpenShiftVersionPodNamespace is a namespace expected to contain pods whose environment includes OpenShift version
+	// information.
+	OpenShiftVersionPodNamespace = "openshift-apiserver-operator"
+	// OpenShiftVersionPodApp is a value for the "app" label to select pods whose environment includes OpenShift version
+	// information.
+	OpenShiftVersionPodApp = "openshift-apiserver-operator"
+
+	// ImageVersionVariable is the environment variable whose value contains the operator image version. For OpenShift
+	// operators, this matches the OpenShift version.
+	ImageVersionVariable = "OPERATOR_IMAGE_VERSION"
+)
+
+// NewOpenShiftVersionProvider creates telemetry data provider that will query the
+// configured k8s cluster - using the provided client - to get cluster k8s version.
```

Comment:
The comment is copy pasted from cluster version provider.

I'd include the information here that the provider doesn't return the openshift version entry at all if it's not detected. And ideally include it in the comment for `openShiftVersionReport` as well.